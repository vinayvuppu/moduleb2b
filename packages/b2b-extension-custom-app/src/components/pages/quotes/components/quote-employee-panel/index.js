export { default } from './quote-employee-panel';

export * from './quote-employee-panel';
