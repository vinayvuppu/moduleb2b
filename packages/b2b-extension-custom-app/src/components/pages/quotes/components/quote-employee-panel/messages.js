import { defineMessages } from 'react-intl';

export default defineMessages({
  panelTitle: {
    id: 'Quote.Details.EmployeePanel.panelTitle',
    defaultMessage: 'Employee',
  },
  goToEmployee: {
    id: 'Quote.Details.EmployeePanel.goToEmployee',
    defaultMessage: 'Go to employee',
  },
  goToCompany: {
    id: 'Quote.Details.EmployeePanel.goToCompany',
    defaultMessage: 'Go to company',
  },
  employeeEmailLabel: {
    id: 'Quote.Details.EmployeePanel.customerEmailLabel',
    defaultMessage: 'Employee email: {email}',
  },
  companylabel: {
    id: 'Quote.Details.EmployeePanel.companylabel',
    defaultMessage: 'Company name: {name}',
  },
});
