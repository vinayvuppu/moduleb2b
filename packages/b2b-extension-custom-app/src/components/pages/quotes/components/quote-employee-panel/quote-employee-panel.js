import React from 'react';
import PropTypes from 'prop-types';
import { useIntl, FormattedMessage } from 'react-intl';
import { Link } from 'react-router-dom';
import {
  Spacings,
  CollapsiblePanel,
  IconButton,
  Tooltip,
  UserLinearIcon,
  ConnectedSquareIcon,
  MailIcon,
  Text,
} from '@commercetools-frontend/ui-kit';

import messages from './messages';

export const QuoteEmployeePanel = props => {
  const intl = useIntl();

  const { formatMessage } = intl;

  // eslint-disable-next-line react/display-name
  const renderLinks = () => (
    <Spacings.Inline>
      <Link
        to={`/${props.projectKey}/b2b-extension/employees/${props.quote.employeeId}`}
        id="employee-panel-employee-link"
      >
        <Tooltip placement="left" title={formatMessage(messages.goToEmployee)}>
          <IconButton
            data-track-component="IconButton"
            data-track-label="go-to-employee"
            data-track-event="click"
            label={formatMessage(messages.goToEmployee)}
            icon={<UserLinearIcon />}
          />
        </Tooltip>
      </Link>
      <Link
        to={`/${props.projectKey}/b2b-extension/companies/${props.quote.company.id}`}
        id="employee-panel-company-link"
      >
        <Tooltip placement="left" title={formatMessage(messages.goToCompany)}>
          <IconButton
            data-track-component="IconButton"
            data-track-label="go-to-company"
            data-track-event="click"
            label={formatMessage(messages.goToCompany)}
            icon={<ConnectedSquareIcon />}
          />
        </Tooltip>
      </Link>
    </Spacings.Inline>
  );

  return (
    <CollapsiblePanel
      data-testid="quote-employee-panel"
      header={
        <CollapsiblePanel.Header>
          <FormattedMessage {...messages.panelTitle} />
        </CollapsiblePanel.Header>
      }
      headerControls={props.isAuthorized && renderLinks()}
    >
      <Spacings.Stack scale="xs">
        <Spacings.Inline
          data-testid="employee-panel-employeeEmail"
          alignItems="center"
        >
          <Text.Detail>
            <FormattedMessage
              {...messages.employeeEmailLabel}
              values={{ email: props.quote.employeeEmail }}
            />
          </Text.Detail>
          <MailIcon />
        </Spacings.Inline>
        <Spacings.Inline
          data-testid="employee-panel-companyName"
          alignItems="center"
        >
          <Text.Detail>
            <FormattedMessage
              {...messages.companylabel}
              values={{ name: props.quote.company.name }}
            />
          </Text.Detail>
        </Spacings.Inline>
      </Spacings.Stack>
    </CollapsiblePanel>
  );
};

QuoteEmployeePanel.displayName = 'QuoteEmployeePanel';
QuoteEmployeePanel.propTypes = {
  quote: PropTypes.object,
  projectKey: PropTypes.string.isRequired,
  isAuthorized: PropTypes.bool.isRequired,
};

export default QuoteEmployeePanel;
