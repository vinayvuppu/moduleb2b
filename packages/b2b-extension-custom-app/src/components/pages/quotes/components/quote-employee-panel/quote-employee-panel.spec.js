import React from 'react';
import { shallow } from 'enzyme';

import { QuoteEmployeePanel } from './quote-employee-panel';

jest.mock('react-intl', () => {
  const actual = jest.requireActual('react-intl');
  return {
    ...actual,
    useIntl: jest.fn(() => ({
      formatMessage: jest.fn(() => 'formatted message'),
    })),
  };
});

const createTestProps = props => ({
  projectKey: 'project-key-1',
  quote: {
    id: 'quote-id',
    employeeId: 'employee-id',
    employeeEmail: 'employee@company.com',
    company: {
      id: 'company-id',
      name: 'company-name',
    },
  },
  isAuthorized: true,
  ...props,
});

describe('rendering', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = createTestProps();
    wrapper = shallow(<QuoteEmployeePanel {...props} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
