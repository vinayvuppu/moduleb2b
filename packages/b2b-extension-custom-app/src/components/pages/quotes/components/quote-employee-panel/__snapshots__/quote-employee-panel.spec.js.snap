// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should match snapshot 1`] = `
<CollapsiblePanel
  condensed={false}
  data-testid="quote-employee-panel"
  header={
    <CollapsiblePanelHeader>
      <FormattedMessage
        defaultMessage="Employee"
        id="Quote.Details.EmployeePanel.panelTitle"
        values={Object {}}
      />
    </CollapsiblePanelHeader>
  }
  headerControls={
    <Inline
      alignItems="flex-start"
      justifyContent="flex-start"
      scale="s"
    >
      <ForwardRef
        id="employee-panel-employee-link"
        to="/project-key-1/b2b-extension/employees/employee-id"
      >
        <ToolTip
          closeAfter={0}
          components={Object {}}
          horizontalConstraint="scale"
          off={false}
          placement="left"
          styles={
            Object {
              "body": Object {},
            }
          }
          title="formatted message"
        >
          <IconButton
            data-track-component="IconButton"
            data-track-event="click"
            data-track-label="go-to-employee"
            icon={<UserLinearIcon />}
            isToggleButton={false}
            label="formatted message"
            shape="round"
            size="big"
            theme="default"
            type="button"
          />
        </ToolTip>
      </ForwardRef>
      <ForwardRef
        id="employee-panel-company-link"
        to="/project-key-1/b2b-extension/companies/company-id"
      >
        <ToolTip
          closeAfter={0}
          components={Object {}}
          horizontalConstraint="scale"
          off={false}
          placement="left"
          styles={
            Object {
              "body": Object {},
            }
          }
          title="formatted message"
        >
          <IconButton
            data-track-component="IconButton"
            data-track-event="click"
            data-track-label="go-to-company"
            icon={<ConnectedSquareIcon />}
            isToggleButton={false}
            label="formatted message"
            shape="round"
            size="big"
            theme="default"
            type="button"
          />
        </ToolTip>
      </ForwardRef>
    </Inline>
  }
  headerControlsAlignment="right"
  horizontalConstraint="scale"
  id="1"
  isDisabled={false}
  theme="dark"
>
  <Stack
    alignItems="stretch"
    scale="xs"
  >
    <Inline
      alignItems="center"
      data-testid="employee-panel-employeeEmail"
      justifyContent="flex-start"
      scale="s"
    >
      <TextDetail>
        <FormattedMessage
          defaultMessage="Employee email: {email}"
          id="Quote.Details.EmployeePanel.customerEmailLabel"
          values={
            Object {
              "email": "employee@company.com",
            }
          }
        />
      </TextDetail>
      <MailIcon />
    </Inline>
    <Inline
      alignItems="center"
      data-testid="employee-panel-companyName"
      justifyContent="flex-start"
      scale="s"
    >
      <TextDetail>
        <FormattedMessage
          defaultMessage="Company name: {name}"
          id="Quote.Details.EmployeePanel.companylabel"
          values={
            Object {
              "name": "company-name",
            }
          }
        />
      </TextDetail>
    </Inline>
  </Stack>
</CollapsiblePanel>
`;
